File: leetcode/python/1.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # make a hashmap to store element and their index
        hashmap = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in hashmap:
                return [hashmap[complement], i]
            hashmap[num] = i


if __name__ == "__main__":
    nums = [2, 7, 11, 15]
    target = 9
    solution = Solution()
    result = solution.twoSum(nums, target)
    print(result)

File: leetcode/go/1.go
package main

func twoSum(nums []int, target int) []int {
	mp := make(map[int]int) //store element-index

	for idx, ele := range nums {
		diff := target - ele
		if idx2, exists := mp[diff]; exists {
			return []int{idx, idx2}
		} else {
			mp[ele] = idx
		}
	}
	return nil
}

func twoSumsAgain(nums []int, target) []int {
	mp := make(map[int]int) //stores element-index
	for idx,val := range nums{
		elem2 := target - val
		if idx2, exists := mp[elem2]; exists{
			return []int{idx,idx2}
		}
	}
	return nil
}

func twoSumsFor(nums []int, target) []int{
	for i := 0; i < len(nums); i++{
		elem2 = target - nums[i]
		for j := i + 1; j <
	}
}
